### DISCUSSION TITLE
Add a configuration for macOS users to explicitly set the NSWindow's Appearance to Dark or Light.

### DISCUSSION THREAD

**Discussion by kennycarruthers** [wezterm/wezterm#5281] [Ideas]
It would be nice to have a configuration option that allows macOS users to explicitly set the appearance of the `NSApp` used by WezTerm rather than only supporting the current appearance of the system. 

For example, I normally have the 'Light' appearance set at the system level, but would prefer to have WezTerm launch in the 'Dark' appearance. It's a minor improvement, but it means that the `NSWindow's` titlebar better matches the dark theme I'm likely using inside the terminal.

(Apple does something like this for their "pro" apps, where they always launch in Dark mode.)

Given there's already `config.macos_window_background_blur`, perhaps `config.macos_application_appearance` or `config.macos_window_appearance` could be considered. 

Apple's documentation for setting the appearance on the application is here: 

https://developer.apple.com/documentation/appkit/nsapplication/2967170-appearance?language=objc

**Comment by pillarcoin** [comment]
https://github.com/wez/wezterm/pull/6521
